datasets/klue/klue.py

```diff
@@ -0,0 +1,521 @@
+                for id_, row in enumerate(f):
+                    # In train file, "source" is written as "genre"
+                    features = {
+                        "guid": row["guid"],
+                        "source": row["source"] if "source" in row else row["genre"],
+                        "premise": row["premise"],
+                        "hypothesis": row["hypothesis"],
+                        "label": row["gold_label"],
+                    }
+                    yield id_, features
+
+        if self.config.name == "ner":
+            with open(data_file, encoding="UTF-8") as f:
+                reader = csv.reader(f, delimiter="\t", quoting=csv.QUOTE_NONE)
+                for _ in range(5):  # skip headers
+                    next(reader)
+                id_ = -1
+                for row in reader:
+                    if row:
+                        if row[0].startswith("##"):
+                            id_ += 1
+                            char, ne_tag = [], []
+                            sentence = row[1]
+                        else:
+                            char.append(row[0])
+                            ne_tag.append(row[1])
+                    else:  # new line
+                        assert len(char) == len(ne_tag)
+                        yield id_, {"sentence": sentence, "char": char, "ne_tag": ne_tag}
+
+        if self.config.name == "dp":
+            with open(data_file, encoding="UTF-8") as f:
+                reader = csv.reader(f, delimiter="\t", quoting=csv.QUOTE_NONE)
+                for _ in range(5):  # skip headers
+                    next(reader)
+                id_ = -1
+                for row in reader:
+                    if row:
+                        if row[0].startswith("##"):
+                            id_ += 1
+                            index = []
+                            word_form = []
+                            lemma = []
+                            pos = []
+                            head = []
+                            deprel = []
+                            sentence = row[1]
+                        else:
+                            index.append(row[0])
+                            word_form.append(row[1])
+                            lemma.append(row[2])
+                            pos.append(row[3])
+                            head.append(row[4])
+                            deprel.append(row[5])
+                    else:  # new line
+                        assert len(index) == len(word_form) == len(lemma) == len(pos) == len(head) == len(deprel)
+                        yield id_, {
+                            "sentence": sentence,
+                            "index": index,
+                            "word_form": word_form,
+                            "lemma": lemma,
+                            "pos": pos,
+                            "head": head,
+                            "deprel": deprel,
+                        }
+
```

Comment:
don't forget the last example if there is one:

```suggestion
                if index:  # last example
                    assert len(index) == len(word_form) == len(lemma) == len(pos) == len(head) == len(deprel)
                    yield id_, {
                        "sentence": sentence,
                        "index": index,
                        "word_form": word_form,
                        "lemma": lemma,
                        "pos": pos,
                        "head": head,
                        "deprel": deprel,
                    }

```

Reply:
I checked it but if this suggestion is added, then last example is added twice.

Reply:
Ok thanks for checking :) you can ignore this suggestion then